File: oo-patterns/src/main/java/victor/training/oo/creational/factory/blocks/Child.java
package victor.training.oo.creational.factory.blocks;

import victor.training.oo.creational.factory.blocks.spi.BlockFactory;
import victor.training.oo.creational.factory.blocks.spi.Board;
import victor.training.oo.creational.factory.blocks.spi.Cube;

public class Child {
	public void playWith(BlockFactory cutie) {
		Board board = cutie.createBoard();
		Cube lastCube = cutie.createCube();
		System.out.println("\"Pun primul cub " + lastCube + " pe tabla " + board);
		for (int i = 2; i <= 10; i++) {
			Cube cube = cutie.createCube();
			System.out.println("\"Iau cubul " + i + ": " + cube + " si il pun peste " + lastCube);
			cube.stackOnto(lastCube);
			lastCube = cube;
		}
		System.out.println("Gata. Asta voi face inca 6 luni de acum inainte... ;p");
	}
}

File: oo-patterns-php/victor/training/oo/creational/abstractfactory/Child.php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: VictorRentea
 * Date: 9/18/2017
 * Time: 12:43 AM
 */

namespace victor\training\oo\creational\abstractfactory;


use victor\training\oo\creational\abstractfactory\spi\BlockFactory;

class Child
{
    public function playWith(BlockFactory $cutie)
    {
        $board = $cutie->createBoard();
        $previousCube = $cutie->createCube();
        printf("Pun primul cub " . $previousCube . " pe tabla " . $board . "\n");
        for ($i = 2; $i <= 10; $i++) {
            $cube = $cutie->createCube();
            printf("\"Iau cubul " . i . ": " . $cube . " si il pun peste " . $previousCube . "\n");
            $cube->stackOnto($previousCube);
            $previousCube = $cube;
        }
        printf("Gata. Asta voi face inca 6 luni de acum inainte... ;p\n");
    }
}